index.bs

```diff
@@ -38,13 +38,64 @@ spec: WEBDRIVER; urlPrefix: https://w3c.github.io/webdriver/
 Introduction {#intro}
 =====================
 
+<em>This section is non-normative.</em>
+
 [[WEBDRIVER|WebDriver]] defines a protocol for introspection and
 remote control of user agents. This specification extends WebDriver by
 introducing bidirectional communication. In place of the strict
 command/response format of WebDriver, this permits events to stream
 from the user agent to the controlling software, better matching the
 evented nature of the browser DOM.
 
+Model {#model}
+==============
+
+This section defines the basic concepts of the WebDriver BiDi
+protocol. These terms are distinct from their representation at the
+<a href=#transport>transport</a> layer, and can be thought of as their
+in-memory representation.
```

Comment:
I'll just drop the "and can be thought of as their in-memory representation" bit. I think it will turn out inconvenient to not have the ID on the command concept, but please comment again if you'd like to start that way and add it when we find it's a problem.

Reply:
So I think I'd like to go a bit further and define the underlying types separately from the messages. So I'd like to have something like:

"""
A `Command Type` has a string `command name` and a set of parameters which are defined by subtypes.

A `Command message` consists of an instance of a `Command Type` and an `id` which is an integer used as a connection-unique identifier for a specific message.
"""

Reply:
I think I agree with where you're going. While there will be some behavior between all commands, there will be no instance of the **command** type, it's an abstract type. Where I've said "any type" is likely to not end up useful, the parameters should as you say be defined by the concrete subtypes.

When we get to defining some concrete command, I think we can start with prose in the style of "`thingHappened` is an **event** in the `Something` domain with parameters **type** (an enum of ...) and **message** (a string)."

I'll give it a shot to move in that direction tomorrow.

Aside: When we've repeated that enough times we might be tempted to look at https://github.com/w3c/webdriver-bidi/issues/21, so I have in the back of my mind that some of this won't be defined with a bunch of prose, just like we don't say "`doThing` is a **method** for the `Something` interface with arguments ..."